src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala

```diff
@@ -425,4 +425,28 @@ class TestErrorAggregator extends AsyncFlatSpec with Matchers with BeforeAndAfte
     val df = ErrorAggregator.aggregate(spark.sqlContext.createDataset(messages).toDF, raiseOnError = false, online = false)
     df.schema.fields.map(_.name) should not contain ("client_id")
   }
+
+  "BuildId" should "not be older than 6 months" in {
+    import spark.implicits._
+    val messages = TestUtils.generateMainMessages(
+      1, Some(Map(
+        "environment.build" -> """{"buildId": "20170602"""",
```

Comment:
These tests are testing the case of a build after a submission date. We should be testing the opposite - a build BEFORE a submission date; specifically: a build < 6 months before a submission_date, and a build > 6 months before a submission date. Any builds after a submission_date should be ignored, since that makes no sense!

Reply:
Good catch, I also found another bug while fixing this 😄